bitswap/client/internal/session/session.go

```diff
@@ -396,12 +409,68 @@
 		// ¯\_(ツ)_/¯
 		return
 	}
+
 	go func(k cid.Cid) {
-		for p := range s.providerFinder.FindProvidersAsync(ctx, k) {
-			// When a provider indicates that it has a cid, it's equivalent to
-			// the providing peer sending a HAVE
-			s.sws.Update(p, nil, []cid.Cid{c}, nil)
+		ctx, span := internal.StartSpan(ctx, "Session.findMorePeers")
+		defer span.End()
+		if span.IsRecording() {
+			span.SetAttributes(attribute.Stringer("cid", c))
+		}
```

Comment:
Can't this be done directly? AFAIK, it won't call `.String` unless it's indeed recording. I think it could indeed be more problematic if directly passed as a string and needed to encode it every time. But here it seems fine. Is there another reason?

Reply:
It literally always call `.String` instantly:
https://github.com/open-telemetry/opentelemetry-go/blob/98b32a6c3a87fbee5d34c063b9096f416b250897/attribute/kv.go#L85
```go
// Stringer creates a new key-value pair with a passed name and a string
// value generated by the passed Stringer interface.
func Stringer(k string, v fmt.Stringer) KeyValue {
	return Key(k).String(v.String())
}
```